lib/shared-functions.sh

```diff
@@ -19,6 +19,46 @@ function read_image_version() {
   IMAGE_VERSION_PATCH=${BASH_REMATCH[3]}
 }
 
+function read_mongo_version() {
+
+  # Reads `MONGO_VERSION` from config/overleaf.rd if to set the correct mongo shell
+  # as $MONGOSH. If `MONGO_VERSION` is not defined it tries to extract the version
+  # from `MONGO_IMAGE`, issuing a warning if it's a non-stock image.
+
+  local mongo_version=$(read_configuration "MONGO_VERSION")
+  if [ -z "${mongo_version}" ]; then
+    local mongo_image=$(read_configuration "MONGO_IMAGE")
+    if [[ "$mongo_image" =~ ^mongo:([0-9]+)\.(.*)$ ]]; then
+      MONGO_VERSION_MAJOR=${BASH_REMATCH[1]}
+    else
+      echo "-------------------------------------------------------"
+      echo "---------------------  WARNING  -----------------------"
+      echo "-------------------------------------------------------"
+      echo "  You are using a non-stock MONGO_IMAGE: $mongo_image"
+      echo ""
+      echo "  When using custom mongo docker images you must specify MONGO_VERSION in your config/overleaf.rc,"
+      echo "  otherwise you may experience problems communicating with your mongo instance."
+      echo "  Example: MONGO_VERSION=6.0"
+      echo "-------------------------------------------------------"
+      echo "---------------------  WARNING  -----------------------"
+      echo "-------------------------------------------------------"
+      MONGO_VERSION_MAJOR=5 # default to version 5 for backwards compatibility
```

Comment:
I would prefer failing here rather than producing a cryptic error down the line.